File: RB_challenges/medium_1/10.rb
class Prime
  def self.nth(rank)
    raise ArgumentError if rank < 1 || rank.class != Integer
    primes = [2]
    candidate = 3
    while primes.length < rank
      primes << candidate unless primes.take_while { |x| x <= Math.sqrt(candidate) }
      .any? { |prime| candidate % prime == 0 }
      candidate += 2
    end
    primes.last
  end
end

File: JS210/practice_problems/other/10.js
let isPrime = (num) => {
  prime = true;

  if (num < 2 || (num !== 2 && num % 2 == 0)) {
    prime = false;
  } else {
    for (let candidate = 3; candidate <= Math.floor(Math.sqrt(num)); candidate += 2) {
      if (num % candidate === 0) prime = false;
    }
  }

  console.log(prime);
}

isPrime(1);   // false
isPrime(2);   // true
isPrime(3);   // true
isPrime(43);  // true
isPrime(55);  // false
isPrime(0);   // false